AppCenter/AppCenter/Internals/Context/UserId/MSUserIdContext.m

```diff
@@ -75,20 +77,30 @@ - (NSString *)userId {
 }
 
 - (void)setUserId:(nullable NSString *)userId {
+  NSArray *synchronizedDelegates;
   @synchronized(self) {
+    BOOL sameUserId = (!userId && !self.currentUserIdInfo.userId) || [self.currentUserIdInfo.userId isEqualToString:(NSString *)userId];
+    self.currentUserIdInfo.timestamp = [NSDate date];
```

Comment:
Why do you update the timestamp here? Do you want to update this value even though it has same userId?
The `timestamp` indicates start time for the userId.

Reply:
exactly. We receinved same userId - so we dont need to add that to history (as pointed out by prev. comments), yet we can update info on when we did get the latest userId, which I did implement in this small refactor

Reply:
We've synced offline for the comment. The timestamp indicates when the userId starts, it shouldn't be updated in that condition. Thanks for addressing the comment.